File: controller-go/api/schema/audio-book.go
package schema

import (
	"time"
)

type AudioBook struct {
	Id          int           `json:"id"`
	Title       string        `json:"title"`
	LastPlayed  time.Time     `json:"lastPlayed"`
	Card        *Card         `json:"card"`
	TimesPlayed int           `json:"timesPlayed"`
	TrackList   []*AudioTrack `json:"trackList"`
}

File: ui/src/app/main/service/audio-book.ts
import {AudioBookTrackList} from './audio-book-track-list';
import {Card} from './card';

export interface AudioBook {
    id: number;
    isMusic: boolean;
    title: string;
    lastPlayed: Date;
    card?: Card;
    timesPlayed: number;
    trackList: AudioBookTrackList[];
}
